### PR TITLE
feat: [SRM-12156]: adding remaining testcases for SLOs listing page

### PR DESCRIPTION
**Pull request by harshilbits**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_NA_** 
 


            
 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/11426)

**Comment by harshilbits** [comment]
retrigger titlecheck

File: src/modules/85-cv/pages/slos/__tests__/CVSLOsListingPage.test.tsx [hunk 1]
```diff
@@ -93,6 +94,36 @@ describe('CVSLOsListingPage', () => {
...
+    })
+  })
+
+  test('Should be able to search the SLO', async () => {
+    const { getByText, container } = render(<ComponentWrapper />)
+    await waitFor(async () => {
+      expect(getByText('SLO-4')).toBeInTheDocument()
+
+      const searchBox = container.querySelector('input[placeholder="cv.slos.searchSLO"]') as HTMLInputElement
+      fireEvent.change(searchBox, { target: { value: 'SLO1' } })
```
Comment (praneshg239) [inline src/modules/85-cv/pages/slos/__tests__/CVSLOsListingPage.test.tsx:119]: always use `userEvent`

File: src/modules/85-cv/pages/slos/__tests__/CVSLOsListingPage.test.tsx [hunk 2]
```diff
@@ -93,6 +94,36 @@ describe('CVSLOsListingPage', () => {
...
+
+      await waitFor(() => {
```
Comment (praneshg239) [inline src/modules/85-cv/pages/slos/__tests__/CVSLOsListingPage.test.tsx:121]: why used nested `waitFor`?

**Review by praneshg239 — COMMENTED** [review]
Please check

File: src/modules/85-cv/pages/slos/__tests__/CVSLOsListingPage.test.tsx (hunk 2 above)
Comment (harshilbits) [reply src/modules/85-cv/pages/slos/__tests__/CVSLOsListingPage.test.tsx:121]: 1. First wait is for listing api.
2. Second wait for -> Once we do a search then api call happens for searched text and we need to wait for the api call to get completed to do further assertions.

File: src/modules/85-cv/pages/slos/__tests__/CVSLOsListingPage.test.tsx (hunk 1 above)
Comment (harshilbits) [reply src/modules/85-cv/pages/slos/__tests__/CVSLOsListingPage.test.tsx:119]: Test suite is getting stuck while using `userEvent `, will need to check.

**Review by praneshg239 — COMMENTED** [review]
please checl

File: src/modules/85-cv/pages/slos/__tests__/CVSLOsListingPage.test.tsx [hunk 3]
```diff
@@ -93,6 +94,36 @@ describe('CVSLOsListingPage', () => {
     jest.spyOn(cvServices, 'useResetErrorBudget').mockReturnValue({ mutate: jest.fn(), loading: false } as any)
   })
 
+  test('Should render created SLOs and display all the fields for the created SLO', async () => {
+    const { getByText } = render(<ComponentWrapper />)
+    await waitFor(() => {
+      expect(getByText('SLO-4')).toBeInTheDocument()
```
Comment (deepesh-ui) [inline src/modules/85-cv/pages/slos/__tests__/CVSLOsListingPage.test.tsx:100]: NIT : can we check values from mockSLODashboardWidgetsData 

expect(getByText(mockSLODashboardWidgetsData.content.environmentName)).toBeInTheDocument()

**Review by deepesh-ui — APPROVED** [review]

**Review by praneshg239 — APPROVED** [review]

File: src/modules/85-cv/pages/slos/__tests__/CVSLOsListingPage.test.tsx (hunk 3 above)
Comment (harshilbits) [reply src/modules/85-cv/pages/slos/__tests__/CVSLOsListingPage.test.tsx:100]: Sure...will do those enhancements in upcoming PRs.

**Pull request merged**